frontend/packages/topology/src/components/graph-view/components/nodes/decorators/UrlDecorator.tsx

```diff
@@ -18,14 +19,17 @@ const UrlDecorator: React.FC<DefaultDecoratorProps> = ({ element, radius, x, y }
   const { t } = useTranslation();
   const resourceObj = getResource(element);
   const url = useRoutesURL(resourceObj);
+  const routeURL = resourceObj?.metadata?.annotations[ROUTE_URL_ANNOTATION];
+  const disabledRoute = resourceObj?.metadata?.annotations[ROUTE_DISABLED_ANNOTATION];
+  const decoratorURL = routeURL || url;
 
-  if (!url) {
+  if (!url || disabledRoute === 'true') {
```

Comment:
per this logic route won't be shown if the default route is not found via `useRoutesURL` and even if the user provides annotations for `ROUTE_URL_ANNOTATION`, is this intended?

Reply:
We allow the user to modify or hide the URL of the URL decorator if the URL is already exists for the workloads.